thentos-purescript/src/Register.purs

```diff
@@ -170,38 +315,199 @@ cl = P.class_ <<< H.className
 
 -- * eval
 
-eval :: forall g. Natural Query (ComponentDSL State Query g)
-eval (KeyPressed updateState next) = do
+eval :: forall eff g. (MonadAff (Effs eff) g)
+    => Natural (Query (Effs eff)) (ComponentDSL (State (Effs eff)) (Query (Effs eff)) g)
+
+eval e@(KeyPressed updateState next) = do
+    liftAff' $ log $ stringify e
```

Comment:
This function combination is called several times, maybe it would make sense to define it a helper function and then just call that one (e.g. `liftLogS e`)?